### PR TITLE
Static analysis fixes

### PR DESCRIPTION
**Pull request by martinetd**
A bunch of very small fixes.

I'd rather not name what tool it was as I'm using the paid version of a tool I have through work (If I say we could use the "free for open-source scan" it's probably enough to guess what it is, but I really hate the web UI and the free version doesn't give access to the command-line tools... So meh.)

If there is any commit worth discussing I can split it out in its own PR to discuss while we merge the rest, but I think it should all be fairly straightforward

### CODE REVIEW

File: backend/session/direct-ipc.c [hunk 1]
```diff
@@ -159,7 +159,8 @@ static void communicate(int sock) {
 			}
 error:
 			send_msg(sock, ret ? -1 : fd, &ret, sizeof(ret));
-			close(fd);
+			if (fd >= 0)
+				close(fd);
```
Comment (SirCmpwn) [inline backend/session/direct-ipc.c]: braces

File: backend/wayland/wl_seat.c [hunk 2]
```diff
@@ -56,6 +56,9 @@ static void pointer_handle_leave(void *data, struct wl_pointer *wl_pointer,
 	}
 
 	struct wlr_wl_output *output = wl_surface_get_user_data(surface);
+	if (output == NULL) {
+		return;
+	}
```
Comment (SirCmpwn) [inline backend/wayland/wl_seat.c]: This can be an assert

File: types/wlr_keyboard.c [hunk 3]
```diff
@@ -156,7 +157,8 @@ void wlr_keyboard_destroy(struct wlr_keyboard *kb) {
 	}
 	xkb_state_unref(kb->xkb_state);
 	xkb_keymap_unref(kb->keymap);
-	close(kb->keymap_fd);
+	if (kb->keymap_fd >= 0)
+		close(kb->keymap_fd);
```
Comment (SirCmpwn) [inline types/wlr_keyboard.c]: braces

**Comment by SirCmpwn** [comment]
LGTM

File: backend/wayland/wl_seat.c (hunk 2 above)
Comment (martinetd) [reply backend/wayland/wl_seat.c]: Right-o, changed in pointer_handle_enter as well. Force pushed along with braces fix in their appropriate commits

File: backend/backend.c [hunk 4]
```diff
@@ -203,19 +203,22 @@ struct wlr_backend *wlr_backend_autocreate(struct wl_display *display,
...
 				wlr_backend_destroy(backend);
 				wlr_session_destroy(session);
+				free(names);
 				return NULL;
 			}
 
 			name = strtok_r(NULL, ",", &saveptr);
 		}
 
+		free(names);
```
Comment (martinetd) [inline backend/backend.c]: (last push added this line, I hadn't looked deeply enough into what `names` was used for and it came up on recheck. Am done rechecking now)

**Comment by emersion** [comment]
Thanks!

**Pull request merged**